src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,327 @@
+
+  @Input()
+  get name(): string {
+    return this._name;
+  }
+
+  set name(value: string) {
+    this._name = value;
+
+    this._updateButtonToggleNames();
+  }
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(newValue: any) {
+    if (this._value != newValue) {
+      this._value = newValue;
+
+      this._updateSelectedButtonToggleFromValue();
+      this._emitChangeEvent();
+    }
+  }
+
+  @Input()
+  get selected() {
+    return this._selected;
+  }
+
+  set selected(selected: MdButtonToggle) {
+    this._selected = selected;
+    this.value = selected ? selected.value : null;
+
+    if (selected && !selected.checked) {
+      selected.checked = true;
+    }
+  }
+
+  private _updateButtonToggleNames(): void {
+    (this._buttonToggles || []).forEach((toggle) => {
+      toggle.name = this._name;
+    });
+  }
+
+  // TODO: Refactor into shared code with radio.
+  private _updateSelectedButtonToggleFromValue(): void {
+    let isAlreadySelected = this._selected != null && this._selected.value == this._value;
+
+    if (this._buttonToggles != null && !isAlreadySelected) {
+      let matchingButtonToggle = this._buttonToggles.filter(
+          buttonToggle => buttonToggle.value == this._value)[0];
+
+      if (matchingButtonToggle) {
+        this.selected = matchingButtonToggle;
+      } else if (this.value == null) {
+        this.selected = null;
+        this._buttonToggles.forEach(buttonToggle => {buttonToggle.checked = false; });
+      }
+    }
+  }
+
+  /** Dispatch change event with current selection and group value. */
+  private _emitChangeEvent(): void {
+    let event = new MdButtonToggleChange();
+    event.source = this._selected;
+    event.value = this._value;
+    this._change.emit(event);
+  }
+}
+
+/** Multiple selection button-toggle group. */
+@Directive({
+  selector: 'md-button-toggle-group[multiple]',
+})
+export class MdButtonToggleGroupMultiple {
+  /** Disables all toggles in the group. */
+  private _disabled: boolean = null;
+
+  /** Child button toggles. */
+  @ContentChildren(forwardRef(() => MdButtonToggle))
+  private _buttonToggles: QueryList<MdButtonToggle> = null;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-button-toggle',
+  templateUrl: 'button-toggle.html',
+  styleUrls: ['button-toggle.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdButtonToggle implements OnInit {
+  /** Whether or not this button toggle is checked. */
+  private _checked: boolean = false;
+
+  /** Type of the button toggle. Either 'radio' or 'checkbox'. */
+  private _type: string = null;
```

Comment:
TypeScript has a cool feature that would be useful here: string literal types. You can do this:
```ts
private _type: 'checkbox' | 'radio';
```
And it will enforce via the compiler that the only values allowed for the property are `checkbox` and `radio`.